docs/rfcs/builder-config-wildcard.md

```diff
@@ -0,0 +1,153 @@
+# OpenTelemetry Wildcard Name Matching
+
+## Overview
+
+### What and Why
+
+The following is a specification for OpenTelemetry Wildcard Name Matching grammar that could be added to the generated `metadata` packages from `mdatagen`. This grammar will facilitate the use of wildcards in metric and resource attribute configurations to enabling/disabling metrics/resource attributes in a more concise way, allowing a configuration like:
+
+```yaml
+metrics:
+  "*":
+    enabled: false
+  system.*:
+    enabled: true
+```
+
+Previously, this would require every metric be listed individually to configure. With this feature, we can allow this to be specified in two simple entries.
+
+### Challenges
+
+The biggest challenge with this feature is that the order specified in the configuration cannot be relied on. Since the underlying type is a Go `map`, order is not deterministic. This is the main reason that I propose coming up with a custom grammar rather than using an existing solution; this specification will include the prioritization system that determines the order of patterns being processed, regardless of the order received.
+
+The grammar being custom means that we will need to manually implement a matching engine. I will include the specification for that as well.
+
+## Specification
+
+### Grammar
+
+The grammar is an extension of the OTel name grammar. In plain english, the OTel name gramma is an OTel identifier (characters with underscores) optionally followed by a `.` and another OTel identifier. With the wildcard matching, the grammar is similar, except instead of a name you can optionally have one of two things:
+
+* A wildcard `*` character, which matches anything in a group. If it is the final group, it will match any number of subsequent groups.
+* A multimatch `{x,y,z,...}` which will match anything specified within it
```

Comment:
When last discussed, this part of the grammar was in contention. Removing it may simplify the implementation a bit.